packages/core/src/messages/responses/leaderboard/leaderboard-get-rank.ts

```diff
@@ -0,0 +1,99 @@
+import {SdkError} from '../../../errors';
+import {
+  ResponseBase,
+  ResponseError,
+  ResponseFound,
+  ResponseNotFound,
+} from '../response-base';
+
+/**
+ * Parent response type for a leaderboard get element rank request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Found}
+ * - {NotFound}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof LeaderboardGetRank.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `LeaderboardGetRank.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {}
+
+class _Found extends Response {
+  private readonly _id: bigint;
+  private readonly _rank: bigint;
+  private readonly _score: number;
+  constructor(id: bigint, rank: bigint, score: number) {
+    super();
+    this._id = id;
+    this._rank = rank;
+    this._score = score;
+  }
+
+  /**
+   * Returns the id of the requested element in the leaderboard
+   * @returns {bigint}
+   */
+  public id(): bigint {
+    return this._id;
+  }
+
+  /**
+   * Returns the rank of the requested element in the leaderboard
+   * @returns {bigint}
+   */
+  public rank(): bigint {
+    return this._rank;
+  }
+
+  /**
+   * Returns the score of the requested element in the leaderboard
+   * @returns {number}
+   */
+  public score(): number {
+    return this._score;
+  }
```

Comment:
This is different from the sorted set GetRank response class, but from the sample integration code, it seemed expected that the user would be able to get id, score, and rank for the requested element, not just rank.